engine/frontend/src/api/container.js

```diff
@@ -21,6 +21,62 @@ const getDataFromApiContainer = async (request, process) => {
     return process(data)
 }
 
+export const getFileArtifactInfo = async(url, fileArtifactName) => {
+    const containerClient = new ApiContainerServicePromiseClient(url);
+    const makeGetFileArtifactInfo = async () => {
+        try {
+            const fileArtifactArgs = new FilesArtifactNameAndUuid()
+            fileArtifactArgs.setFilename(fileArtifactName)
+            const inspectFileArtifactReq = new InspectFilesArtifactContentsRequest()
+            inspectFileArtifactReq.setFileNamesAndUuid(fileArtifactArgs)
+            const responseFromGrpc = await containerClient.inspectFilesArtifactContents(inspectFileArtifactReq, {})
+            return responseFromGrpc.toObject()
+        } catch (error) {
+            return {
+                files:[]
+            }
+        }
+    }
+
+    const processFileArtifact = (data) => {
```

Comment:
this potentially can be done on the backend - @victorcolombo 
What I am doing is - i am converting the list of files to look like
```
{
  dir : {
     sub_dir1: {
            sub_dir2: {
                    file_name: file
                }
        }
  }
}
```

This allows me to show file structure; that is i will only the sub_dir1 on the ui - only after user clicks the parent folder.

Reply:
Good call, do you mind create a ticket for that?